src/extension.ts

```diff
@@ -26,14 +26,19 @@ import { activate as activateSourceKitLSP } from './sourcekit-lsp/extension';
 export async function activate(context: vscode.ExtensionContext) {
 	console.debug('Activating Swift for Visual Studio Code...');
 
-	await activateSourceKitLSP(context);
-
 	const workspaceContext = new WorkspaceContext(context);
+	context.subscriptions.push(workspaceContext);
```

Comment:
The workspace context is added here because the extension initialisation may fail after it has been created. We should be sure it is in the subscription list so `dispose` is called on it.